### PR TITLE
buildkit socket: Add systemd unit

### LINKED ISSUE #9947: Start the buildkit daemon on demand (socket-activated)
Buildkit uses a server (daemon), so trying to use the buildkit client fails if it is not started:

`error: failed to get status: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /run/buildkit/buildkitd.sock: connect: no such file or directory"`

See #9640 (and https://github.com/moby/buildkit#quick-start)

Currently the `buildkitd` has to be started manually, as there is no systemd unit for it.

```console
$ sudo -b buildkitd
$ INFO[0000] auto snapshotter: using overlayfs            
WARN[0000] using host network as the default            
INFO[0000] found worker "p6bbje3hwqqbczwx909fjnaed", labels=map[org.mobyproject.buildkit.worker.executor:oci org.mobyproject.buildkit.worker.hostname:minikube org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/386] 
WARN[0000] using host network as the default            
INFO[0000] found worker "kgulhjepnr8a15fqgv0q4uag7", labels=map[org.mobyproject.buildkit.worker.executor:containerd org.mobyproject.buildkit.worker.hostname:minikube org.mobyproject.buildkit.worker.snapshotter:overlayfs], platforms=[linux/amd64 linux/386] 
INFO[0000] found 2 workers, default="p6bbje3hwqqbczwx909fjnaed" 
WARN[0000] currently, only the default worker can be used. 
INFO[0000] running server on /run/buildkit/buildkitd.sock
```

We should add a "buildkit.service" file for it, so that it can started/stopped/logged with systemd.

Similar to the current "containerd.service"

----

It would be nice if it was started automatically, using a systemd socket-activated service.

`sudo systemctl enable buildkit.socket`
`sudo systemctl start buildkit.socket`

This also allows setting group permissions, so that the socket can be accessed over a tunnel.

```
[Socket]
SocketMode=0660
SocketUser=root
SocketGroup=buildkit
```

Locally one can use `sudo`, but that doesn't work when using ssh tunnels to the unix socket.

`unix:///run/buildkit/buildkitd.sock`

`error: failed to get status: rpc error: code = Unavailable desc = connection error: desc = "transport: error while dialing: dial unix /run/buildkit/buildkitd.sock: connect: permission denied"`

Similar to the other runtimes:

* docker.sock #9229 #9232 
* podman.sock #9230 

Opened issues upstream:

* ~~https://github.com/moby/buildkit/issues/1923~~ 
* https://github.com/moby/buildkit/issues/1793

See http://0pointer.de/blog/projects/socket-activated-containers.html

### LINKED ISSUE #12264: Add configuration for buildkit, in order to not have to use flags
We can pass the setup as configuration, rather than flags.

That makes it easier to start from systemd units, and so on.

We want to use the containerd worker, by default.

`-oci-worker false --containerd-worker true `

And store the result in the "k8s.io" namespace.

`--containerd-worker-namespace k8s.io`

Helps with #9947

### PR DESCRIPTION
**Pull request by spowelljr**
Closes #9947
Closes #12264

Creating a systemd unit that we can use to start the buildkit daemon.

### CODE REVIEW

File: deploy/kicbase/Dockerfile
Comment (afbjorklund) [inline deploy/kicbase/Dockerfile:194]: I think needs to be given twice ?

```
adduser docker podman
adduser docker buildkit
```

Also make sure to create the group.

```
addgroup --system buildkit
```

At the `echo "Installing buildkit ...` block (line 135)

File: deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service
Comment (afbjorklund) [inline deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service:8]: This should probably be `/usr/sbin/buildkitd`, same as in the install command.

**Review by afbjorklund — CHANGES_REQUESTED** [review]

**Review by afbjorklund — APPROVED** [review]
Looking good! Need to try it now, see if we missed something else

File: deploy/kicbase/Dockerfile
Comment (afbjorklund) [inline deploy/kicbase/Dockerfile:129]: Almost forgot, you also have to download and install the systemd unit files: (to /usr/local/lib/systemd/system/)
https://raw.githubusercontent.com/moby/buildkit/v0.9.0/examples/systemd/buildkit.service
https://raw.githubusercontent.com/moby/buildkit/v0.9.0/examples/systemd/buildkit.socket
Too bad there is no decent packaging (like a .deb) for buildkit... (See #10068)

**Review by afbjorklund — CHANGES_REQUESTED** [review]

File: deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service
Comment (afbjorklund) [inline deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service:8]: Eventually these flags can go in /etc/buildkit instead

File: deploy/kicbase/Dockerfile
Comment (afbjorklund) [reply deploy/kicbase/Dockerfile:129]: We want to to the minikube changes as overrides, so that we can leave the upstream files intact (see podman).

https://wiki.archlinux.org/title/Systemd#Drop-in_files

Comment (afbjorklund) [reply deploy/kicbase/Dockerfile:129]: Eventually the /etc/buildkit will have to be populated here as well

File: deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service
Comment (afbjorklund) [reply deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service:8]: `ExecStart=/usr/sbin/buildkitd --addr fd://`

Comment (afbjorklund) [reply deploy/iso/minikube-iso/package/buildkit-bin/buildkit.service:8]: We _could_ use /usr/bin (like docker), even if it is "wrong"

File: deploy/kicbase/Dockerfile
Comment (spowelljr) [reply deploy/kicbase/Dockerfile:146]: Thanks for the guidance @afbjorklund! I implemented downloading and installing the systemd unit files and added the `/etc/buildkit` for kicbase as well, I tested the changes and successfully built the image using `image build`. I created the `/etc/buildkit` file by echoing to a file, I'm not sure if there was somewhere specific you want to put the file and copy it over so please take a look at my changes and let me know.

But the gist is `image build` via kicbase was successful, onto iso.

**Comment by spowelljr** [comment]
Tested and working, waiting for https://github.com/kubernetes/minikube/pull/12425 to be merged before building ISO and kicbase to prevent merge conflicts

**Comment by spowelljr** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
ok-to-build-iso

**Comment by spowelljr** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
ok-to-build-image

**Comment by spowelljr** [comment]
Blocked by https://github.com/cri-o/cri-o/issues/5343

**Comment by spowelljr** [comment]
ok-to-build-image

**Review by sharifelgamal — APPROVED** [review]

**Comment by spowelljr** [comment]
ok-to-build-iso

**Pull request merged**
